src/services/preProcess.ts

```diff
@@ -345,6 +345,24 @@ namespace ts {
                     break;
                 }
 
+                if (scanner.getToken() === SyntaxKind.TemplateHead) {
+                    loop: while (true) {
+                        switch (scanner.getToken()) {
+                            case SyntaxKind.EndOfFileToken:
+                                break loop;
+                            case SyntaxKind.ImportKeyword:
+                                tryConsumeImport();
+                                break;
+                            case SyntaxKind.CloseBraceToken: {
+                                scanner.reScanTemplateToken(/*isTaggedTemplate*/ false);
+                                break loop;

```

Comment:
Okay, here's where I think this won't work, and really a counter isn't technically enough.

```ts

foo `${ funcCall({ prop: 100 }, import("module")) }`

```

Reply:
@DanielRosenwasser I've added a test for this case
```ts
"foo`${ fn({ a: 100 }, import(\"a\"), `${import(\"b\")}`, import(\"c\"), `${import(\"d\")} foo`, import(\"e\")) }`"
```

Reply:
@DanielRosenwasser Do you have additional cases that need to be covered by tests? I would be grateful if you could point me to them.